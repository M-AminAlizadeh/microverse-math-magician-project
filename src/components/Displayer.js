import PropTypes from "prop-types";

const Displayer = ({ obj }) => {
  if (obj === undefined || obj === null) {
    return null; // or return an appropriate fallback component
  } else {
    const show = (obj) => {
      let len = Object.keys(obj).length;
      if (len == "3") {
        if (obj.operation == null && obj.next == null) {
          return obj.total;
        } else if (obj.operation && obj.next == null) {
          return obj.total + " " + obj.operation;
        } else if (obj.operation && obj.next && obj.total) {
          return obj.total + " " + obj.operation + " " + obj.next;
        }
      }
      if (len == "2") {
        return obj.next;
      }
    };

    return <div className="calculator-display number">{show(obj) || 0}</div>;
  }
};

Displayer.propTypes = {
  obj: PropTypes.object.isRequired,
};

export default Displayer;
