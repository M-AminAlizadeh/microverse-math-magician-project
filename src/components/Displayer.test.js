import React from "react";
import renderer from "react-test-renderer";
import Displayer from "./Displayer";

describe("Displayer", () => {
  it("renders correctly", () => {
    const component = renderer.create(<Displayer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
