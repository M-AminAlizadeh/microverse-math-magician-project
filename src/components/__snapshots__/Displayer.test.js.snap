// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displayer renders correctly 1`] = `null`;
